File: src/problems/2108.ts
import fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "/input.txt";
var input = fs.readFileSync(__dirname + filePath).toString();

export const solution = (stdinInput: string) => {
  let [N, ...arr] = stdinInput.trim().split(/\s/).map(Number);

  arr.sort((a, b) => a - b);

  let avg = Math.round(arr.reduce((acc, cur) => acc + cur) / N) + 0;

  let mid = arr[arr.length >> 1];

  let range = Math.abs(arr[arr.length - 1] - arr[0]);

  const freq = new Map<number, number>();
  let maxFreq = 0;

  for (const num of arr) {
    const count = (freq.get(num) || 0) + 1;
    freq.set(num, count);
    maxFreq = Math.max(maxFreq, count);
  }

  const modes = Array.from(freq)
    .filter(([, count]) => count === maxFreq)
    .map(([num]) => num);

  const mode = modes.length === 1 ? modes[0] : modes[1];

  process.stdout.write([avg, mid, mode, range].join("\n"));
};

solution(input);

File: src/problems/2108.py
import sys

input = sys.stdin.readline

N = int(input())
num_list = [int(input()) for _ in range(N)]
di = {}
for i in num_list:
    try:
        di[i] += 1
    except:
        di[i] = 1

avg = round(sum(num_list) / N)
print(avg)

sorted_list = sorted(num_list)
mid = sorted_list[int(len(sorted_list) / 2)]
print(mid)

max_value = max(list(di.values()))
max_list = [k for k, v in di.items() if v == max_value]
if len(max_list) > 1:
    print(sorted(max_list)[1])
else:
    print(sorted(max_list)[0])

range_value = max(num_list) - min(num_list)
print(range_value)
